File: src/main/java/frc/team5190/robot/OI.java
package frc.team5190.robot;

import edu.wpi.first.wpilibj.XboxController;
import edu.wpi.first.wpilibj.buttons.JoystickButton;

@SuppressWarnings("unused")
public class OI {
    private XboxController xbox = new XboxController(0);

    OI() {
        JoystickButton buttonA = new JoystickButton(xbox, 1);
        JoystickButton buttonX = new JoystickButton(xbox, 3);
        JoystickButton buttonB = new JoystickButton(xbox, 2);
    }

    public XboxController getXbox() {
        return xbox;
    }
}

File: src/main/kotlin/frc/team5190/robot/OI.kt
/*
 * Copyright (c) 2018 FRC Team 5190
 * Ryan Segerstrom, Prateek Machiraju
 */

@file:Suppress("unused")

package frc.team5190.robot

import edu.wpi.first.wpilibj.*
import edu.wpi.first.wpilibj.buttons.JoystickButton
import frc.team5190.robot.climb.DeployHookCommand

object Bongos : Joystick(0) {
    fun getLeftBongoSpeed() = when {
        this.getRawButton(4) -> 0.6
        this.getRawButton(2) -> -0.6
        else -> 0.0
    }

    fun getRightBongoSpeed() = when {
        this.getRawButton(3) -> 0.6
        this.getRawButton(1) -> -0.6
        else -> 0.0
    }
}


/**
 * Xbox Controller object
 */
object MainXbox : XboxController(0)

fun XboxController.getLeftX() = getX(GenericHID.Hand.kLeft)
fun XboxController.getLeftY() = getY(GenericHID.Hand.kLeft)

fun XboxController.getRightX() = getX(GenericHID.Hand.kRight)
fun XboxController.getRightY() = getY(GenericHID.Hand.kRight)

fun XboxController.getTriggerPressed(hand: GenericHID.Hand, amount: Double = 0.5) = getTriggerAxis(hand) > amount